src/libm-tester/iutsimd.c

```diff
@@ -176,68 +248,74 @@ typedef Sleef___m256_2 vfloat2;
 #endif
 
 #ifdef ENABLE_VSX
+#include "renamevsx.h"
+#if !defined(USE_INLINE_HEADER)
 #define CONFIG 1
 #include "helperpower_128.h"
-#include "renamevsx.h"
 typedef Sleef_vector_double_2 vdouble2;
 typedef Sleef_vector_float_2 vfloat2;
 #endif
+#endif
 
 #ifdef ENABLE_VSXNOFMA
+#include "renamevsxnofma.h"
+#if !defined(USE_INLINE_HEADER)
 #define CONFIG 2
 #include "helperpower_128.h"
-#include "renamevsxnofma.h"
 typedef Sleef_vector_double_2 vdouble2;
 typedef Sleef_vector_float_2 vfloat2;
 #endif
+#endif
 
 #ifdef ENABLE_PUREC_SCALAR
+#include "renamepurec_scalar.h"
+#if !defined(USE_INLINE_HEADER)
 #define CONFIG 1
 #include "helperpurec_scalar.h"
-#include "renamepurec_scalar.h"
 typedef Sleef_double_2 vdouble2;
 typedef Sleef_float_2 vfloat2;
 #endif
+#endif
 
 #ifdef ENABLE_PURECFMA_SCALAR
+#include "renamepurecfma_scalar.h"
+#if !defined(USE_INLINE_HEADER)
 #define CONFIG 2
 #include "helperpurec_scalar.h"
-#include "renamepurecfma_scalar.h"
 typedef Sleef_double_2 vdouble2;
 typedef Sleef_float_2 vfloat2;
 #endif
+#endif
 
 //
 
 #ifdef ENABLE_DP
-int check_featureDP() {
-  if (vavailability_i(1) == 0) return 0;
+int check_featureDP(double d) {
```

Comment:
Is this needed for the header functionality? Same question for the float version below.

Reply:
With the headers, functions like this have greater tendency of being optimized away.
This change is needed to prevent optimizer from removing this function.